src/integer_mod_q/mat_zq/sort.rs

```diff
@@ -0,0 +1,336 @@
+use crate::{
+    error::MathError,
+    traits::{GetNumColumns, GetNumRows},
+};
+
+impl MatZq {
+    /// Sorts the columns of the matrix based on some condition defined by `cond_func`.
+    ///
+    /// This condition is usually a norm with the described input-output behaviour.
+    ///
+    /// Parameters:
+    /// - `cond_func`: computes values implementing [`Ord`] over the columns of the specified matrix.
+    /// These values are then used to re-order / sort the rows of the matrix.
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// A [`MathError`] is returned if the execution of `cond_func` returned an error.
+    ///
+    /// # Examples
+    /// ## Use a build-in function as condition
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    /// let mat = MatZq::from_str("[[3,2,1]] mod 7").unwrap();
+    /// let cmp = MatZq::from_str("[[1,2,3]] mod 7").unwrap();
+    ///
+    /// let sorted = mat.sort_by_column(MatZq::norm_eucl_sqrd).unwrap();
+    ///
+    /// assert_eq!(cmp, sorted);
+    /// ```
+    /// ## Use a custom function as condition
+    /// This function needs to take a column vector as input and output a type implementing [`PartialOrd`]
+    /// ```
+    /// use qfall_math::{integer_mod_q::MatZq, integer::Z, error::MathError, traits::{GetNumRows, GetEntry}};
+    /// use std::str::FromStr;
+    /// let mat = MatZq::from_str("[[3,2,1]] mod 7").unwrap();
+    /// let cmp = MatZq::from_str("[[1,2,3]] mod 7").unwrap();
+    ///
+    /// fn custom_cond_func(matrix: &MatZq) -> Result<Z, MathError> {
+    ///     let mut sum = Z::ZERO;
+    ///     for row in 0..matrix.get_num_rows() {
+    ///         let entry: Z = matrix.get_entry(row, 0)?;
+    ///         sum = sum + entry;
+    ///     }
+    ///     Ok(sum)
+    /// }
+    ///
+    /// let sorted = mat.sort_by_column(custom_cond_func).unwrap();
+    ///
+    /// assert_eq!(cmp, sorted);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of the same type as `cond_func` if the execution of `cond_func` fails.
+    pub fn sort_by_column<T>(
+        &self,
+        cond_func: fn(&Self) -> Result<T, MathError>,
+    ) -> Result<Self, MathError>
+    where
+        T: Ord,
+    {
+        let mut condition_values = vec![];
+        for col in 0..self.get_num_columns() {
+            condition_values.push(cond_func(&self.get_column(col).unwrap())?);
+        }
+
+        let mut id_vec: Vec<usize> = (0..self.get_num_columns() as usize).collect();
+        id_vec.sort_by_key(|x| &condition_values[*x]);
+
+        let mut out =
+            Self::new(self.get_num_rows(), self.get_num_columns(), self.get_mod()).unwrap();
+        for (col, item) in id_vec.iter().enumerate() {
+            out.set_column(col, self, *item).unwrap();
+        }
+
+        Ok(out)
+    }
+
+    /// Sorts the rows of the matrix based on some condition defined by `cond_func`.
```

Comment:
Same here. I am not quite shure If I missed it in the other PR, if so, please update the MatZ sort